src/stcal/outlier_detection/utils.py

```diff
@@ -85,43 +85,32 @@ def compute_weight_threshold(weight, maskpt):
 
 
 def _abs_deriv(array):
-    """
-    Do not use this function.
-
-    Take the absolute derivative of a numpy array.
-
-    This function assumes off-edge pixel values are 0
-    and leads to erroneous derivative values and should
-    likely not be used.
-    """
-    tmp = np.zeros(array.shape, dtype=np.float64)
-    out = np.zeros(array.shape, dtype=np.float64)
-
-    tmp[1:, :] = array[:-1, :]
-    tmp, out = _absolute_subtract(array, tmp, out)
-    tmp[:-1, :] = array[1:, :]
-    tmp, out = _absolute_subtract(array, tmp, out)
-
-    tmp[:, 1:] = array[:, :-1]
-    tmp, out = _absolute_subtract(array, tmp, out)
-    tmp[:, :-1] = array[:, 1:]
-    tmp, out = _absolute_subtract(array, tmp, out)
-
+    """Take the absolute derivate of a numpy array."""
+    out = np.zeros_like(array)  # use same dtype as input
+
+    # make output values nan where input is nan (for floating point input)
+    if np.issubdtype(array.dtype, np.floating):
+        out[np.isnan(array)] = np.nan
+
+    # compute row-wise absolute diffference
+    d = np.abs(np.diff(array, axis=0))
```

Comment:
Please avoid using one letter variable names (one reason is that when running `ipdb` the letter 'd' has a meaning within that framework).  Maybe `row_diff` for this variable.

Reply:
Thanks! How does 70f4045 look for variable renaming?